src/audio/volume/volume.c

```diff
@@ -157,14 +158,53 @@ static struct comp_dev *volume_new(struct sof_ipc_comp *comp)
 	schedule_task_init(&cd->volwork, SOF_SCHEDULE_LL, SOF_TASK_PRI_MED,
 			   vol_work, dev, 0, 0);
 
-	/* set the default volumes */
+	/* Set the default volumes. If IPC sets min_value or max_value to
+	 * not-zero, use them. Otherwise set to internal limits and notify
+	 * ramp step calculation about assumed range with the range set to
+	 * zero.
+	 */
+	if (vol->min_value || vol->max_value) {
+		if (vol->min_value < VOL_MIN) {
+			/* Use VOL_MIN instead, no need to stop new(). */
+			cd->vol_min = VOL_MIN;
+			trace_volume_error("volume_new(): vol->min_value "
+					   "was limited to VOL_MIN.");
+		} else {
+			cd->vol_min = vol->min_value;
+		}
+
+		if (vol->max_value > VOL_MAX) {
+			/* Use VOL_MAX instead, no need to stop new(). */
+			cd->vol_max = VOL_MAX;
+			trace_volume_error("volume_new(): vol->max_value "
+					   "was limited to VOL_MAX.");
+		} else {
+			cd->vol_max = vol->max_value;
+		}
+
+		cd->vol_ramp_range = vol->max_value - vol->min_value;
+	} else {
+		/* Legacy mode, set the limits to firmware capability where
+		 * VOL_MAX is a very large gain to avoid restricting valid
+		 * requests. The default ramp rate will be computed based
+		 * on 0 - 1.0 gain range assumption when vol_ramp_range
+		 * is set to 0.
+		 */
+		cd->vol_min = VOL_MIN;
+		cd->vol_max = VOL_MAX;
+		cd->vol_ramp_range = 0;
+	}
+
 	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
-		cd->volume[i]  =  MAX(MIN(VOL_MAX, VOL_ZERO_DB), VOL_MIN);
+		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
+				      cd->vol_min);
 		cd->tvolume[i] =  cd->volume[i];
 	}
 
 	trace_volume("vol->initial_ramp = %d, vol->ramp = %d",
 		     vol->initial_ramp, vol->ramp);
+	trace_volume("vol->min_value = %d, vol->max_value = %d",
```

Comment:
@singalsu Could we merge them? One trace log can handle up to 4 arguments, so we should be good here.

Reply:
@tlauda I have problems with my computer to view traces with very long lines. I do this to keep the trace width something that fits my Ubuntu 1080p desktop with default font and console window stretched to max width. Is there a solution to make the sof-logger lines shorter by e.g. omitting or condensing other fields?

But yes sure I can combine these.

Reply:
Done.